lib/events/mock.go

```diff
@@ -26,20 +26,27 @@ import (
 	"github.com/gravitational/trace"
 )
 
+// NewMockAuditLog returns an instance of MockAuditLog.
 func NewMockAuditLog(capacity int) *MockAuditLog {
 	return &MockAuditLog{
 		SlicesC:         make(chan *SessionSlice, capacity),
 		FailedAttemptsC: make(chan *SessionSlice, capacity),
 	}
 }
 
+// Events holds the event type and event fields.
+type Events struct {
+	EventType Event
+	Fields    EventFields
+}
+
 // MockAuditLog is audit log used for tests
 type MockAuditLog struct {
 	sync.Mutex
-	returnError        error
-	FailedAttemptsC    chan *SessionSlice
-	SlicesC            chan *SessionSlice
-	MockEmitAuditEvent func(event Event, fields EventFields) error
+	returnError     error
+	FailedAttemptsC chan *SessionSlice
+	SlicesC         chan *SessionSlice
+	EmittedEvents   *Events
```

Comment:
This might need to be a slice in the future, if an action emits multiple events. OK to keep as is for now.
Change `EmittedEvents/Events` to `EmittedEvent/Event` though.

Reply:
there will be a naming conflict, because `Event` is:
```
type Event struct {
	// Name is the event name.
	Name string
	// Code is the unique event code.
	Code string
}
```

Reply:
got it.
from API design perspective, giving users an exported field (`EmitteEvent`) of an unexported type (`event`) is not desirable. User wouldn't be able to see the definition of `type event` in package docs.
how about `type EmittedEvent` ?

Reply:
ah i see! revised to `type EmittedEvent`